atom/renderer/atom_renderer_client.cc

```diff
@@ -129,6 +130,9 @@ void AtomRendererClient::RenderFrameCreated(
     content::RenderFrame* render_frame) {
   new PepperHelper(render_frame);
   new AtomRenderFrameObserver(render_frame, this);
+
+  // Allow file scheme to handle service worker by default.
+  blink::WebSecurityPolicy::registerURLSchemeAsAllowingServiceWorkers("file");
```

Comment:
we have `webframe::registerURLSchemeAsPrivileged` to do that. So users who need custom schemes to handle service workers have to use both these api to allow on both render and browser side.